amida-tech/hl7-telescoper-client
# Hl7 14 signup login

# What does this do?
* organizes routing (feedback welcome!)
* adds axios for api calls
* adds mobx for state management
* creates userStore for user state and actions
* creates signup form
* changes all components to be named exports
* changes primary color to `blueGrey` 🤷‍♂ 
* disables button ripples globally (personal preference, we can change back if you want)
* makes signUp call to api
* creates `telescoperApi` service object

# Known shortcomings
* no loading state during signup
* ~~the pattern for `flow`s in mobx + typescript is slightly cumbersome, not sure if there's a better way~~
  * ~~possibly one of the other patterns for async actions from the mobx docs would turn out better~~
* I've hardcoded the url for the api as localhost, we need to decide on a configuration pattern
* ~~I've hardcoded the signup endpoint inside the mobx `flow`, we should consider factoring out api calls into a service object that provides a well defined interface for our backend~~

# How to test
**Prereq: https://github.com/amida-tech/hl7-telescoper-api-server/pull/6 should be running locally**
* stand up the project 
* load the site
* it should redirect you to `/auth/login`
* navigate to `/auth/signup` via the url
* see the pretty signup page
* fill it out (there is no client side validation)
* click "Sign Up" with invalid data
* see an error below the form
* click "Sign Up" with valid data
* be navigated to the files page
